rust/src/lib/nispor/alt_name.rs

```diff
@@ -0,0 +1,364 @@
+                    format!(
+                        "Failed to create folder \
+                         {SYSTEMD_NETWORK_LINK_FOLDER}: {e}"
+                    ),
+                )
+            })?;
+    }
+    let iface_name = des_base_iface.name.as_str();
+
+    let match_rules = match des_base_iface.identifier.as_ref() {
+        Some(InterfaceIdentifier::MacAddress) => {
+            let mut match_rules = if let Some(mac) =
+                des_base_iface.permanent_mac_address.as_ref()
+            {
+                format!("PermanentMACAddress={mac}")
+            } else if let Some(mac) = des_base_iface.mac_address.as_ref() {
+                format!("MACAddress={mac}")
+            } else {
+                return Err(NmstateError::new(
+                    ErrorKind::Bug,
+                    format!(
+                        "Got no MAC address for `identifier:mac-address` on \
+                         interface {}, should be failed by previous checker",
+                        des_base_iface.name
+                    ),
+                ));
+            };
+            // Include driver in match rule in case we have two NIC holding the
+            // same MAC. (e.g. M$ Azure - Microsoft Azure Network Adapter)
+            if let Some(cur_base_iface) = cur_base_iface.as_ref() {
+                if let Some(driver) = cur_base_iface.driver.as_ref() {
+                    write!(match_rules, "\nDriver={driver}").ok();
+                }
+            }
+            match_rules
+        }
+        Some(InterfaceIdentifier::PciAddress) => {
+            if let Some(pci_addr) = des_base_iface.pci_address.as_ref() {
+                format!("Path=pci-{pci_addr}")
+            } else {
+                return Err(NmstateError::new(
+                    ErrorKind::Bug,
+                    format!(
+                        "Got no PCI address for `identifier:pci-address` on \
+                         interface {}, should be failed by previous checker",
+                        des_base_iface.name
+                    ),
+                ));
+            }
+        }
+        _ => {
+            gen_systemd_network_link_match_rules(des_base_iface, cur_base_iface)
+        }
+    };
+
+    let file_path = gen_systemd_link_file_path(iface_name);
+    let mut content = {
+        #[cfg_attr(any(), rustfmt::skip)]
+        format!(
+            "{PERSIST_GENERATED_BY}\n\
+            [Match]\n\
+            {match_rules}\n\n\
+            [Link]\n"
+        )
+    };
+    let alt_names: Vec<&str> = alt_names.iter().map(|s| s.as_str()).collect();
+    for alt_name in alt_names.iter() {
+        writeln!(content, "AlternativeName={alt_name}").ok();
+    }
+    // Since systemd stop using other naming policy after matched our link file
+    // we should explicitly name this interface to current interface name,
+    // otherwise, reboot will get NIC named as eth0.
+    writeln!(content, "Name={iface_name}").ok();
```

Comment:
True, but this also means that users can create their own .link files that will take precedence to ours, making it useless.

Plus this:
> Note that some distributions may incorporate .link files in their early boot facilities (e.g. by including copies of the .link files in initramfs). As such it may be necessary to take manual steps to ensure that any local changes are consistent with early-boot storage facilities.

Now I'm even less sure that this is the right approach.

Reply:
This feature is used by OCP where they know how to handle this systemd link priority stuff.

I don't want to waste time to beg NM to support this after second rejection.